p2p/simulations/mocker_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Package simulations simulates p2p networks.
+// A mokcer simulates starting and stopping real nodes in a network.
+package simulations
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestMocker(t *testing.T) {
+	//start the simulation HTTP server
+	_, s := testHTTPServer(t)
+	defer s.Close()
+
+	//create a client
+	client := NewClient(s.URL)
+
+	//start the network
+	err := client.StartNetwork()
+	if err != nil {
+		t.Fatalf("Could not start test network: %s", err)
+	}
+
+	//get the list of available mocker types
+	resp, err := http.Get(s.URL + "/mocker")
+	if err != nil {
+		t.Fatalf("Could not get mocker list: %s", err)
+	}
+
+	if resp.StatusCode != 200 {
+		t.Fatalf("Invalid Status Code received, expected 200, got %d", resp.StatusCode)
+	}
+
+	//check the list is at least 1 in size
+	mockerlist := make([]string, 10)
+	json.NewDecoder(resp.Body).Decode(&mockerlist)
```

Comment:
Check the error, and just pass an uninitialized slice (i.e. `var mockerList []string`)